src/main/java/com/bankbazaar/test/service/EmployeeService.java

```diff
@@ -0,0 +1,117 @@
+package com.bankbazaar.test.service;
+
+
+import com.bankbazaar.test.entity.Department;
+import com.bankbazaar.test.entity.Employee;
+import com.bankbazaar.test.repository.EmployeeRepository;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+@Service
+public class EmployeeService {
+
+    @Autowired
+    private EmployeeRepository employeeRepository;
+    private EmployeeRepository departmentRepository;
+    /**
+     * Insert to Employee table
+     * @param data
+     */
+    public Employee saveEmployee(Employee data)
+    {
+        try {
+            /**
+             * Get department data
+             */
+            final String uriGet = "http://localhost:8080/departmentById/"+data.getDepartmentId().getId();
+            HttpHeaders headers = new HttpHeaders();
+            headers.add("Authorization", "Basic YWRtaW46YWRtaW4xMjM0");
+            HttpEntity request = new HttpEntity(headers);
+
+            ResponseEntity<String> response = new RestTemplate().exchange(uriGet, HttpMethod.GET, request, String.class);
+
+            JsonObject departmentData = JsonParser.parseString(response.getBody()).getAsJsonObject();
+            /**
+             * Store new employee count
+             */
+            int employeeCount = departmentData.get("employeeCount").getAsInt()+1;
+
+            /**
+             * Update department details
+             */
+            final String uriPut = "http://localhost:8080/updateDepartment";
+
+            Department department =new Department();
+            department.setId(departmentData.get("id").getAsInt());
+            department.setEmployeeCount(employeeCount);
+            department.setName(departmentData.get("name").getAsString());
+
+            HttpEntity<Department> update = new HttpEntity<Department>(department,headers);
+
+            new RestTemplate().exchange(uriPut, HttpMethod.PUT, update, String.class);
+
+        }
+        catch (Exception e)
+        {
+            System.out.println("The provided department ID doesn't exist");
```

Comment:
Dont use System.out.println in projects. Always use logs.
Look into @Slf4j annotation

Reply:
okay. however i have removed this part of the code for now.